static_src/main.js

```diff
@@ -177,12 +185,11 @@ var onDelete = function(event, state) {
                 id: state.id,
             }),
             credentials: 'same-origin',
-        })
-        .then(extractJSON)
-        .then(updateModel)
-        .then(function(model) {
-            history.pushState(null, null, '#!list');
-            return onNavigate(null, Object.assign({}, state, model));
+        }).then(function () {
+            return (updateModel(state)).then(function(model) {
+                history.pushState(null, null, '#!list');
+                return onNavigate(null, Object.assign({}, state, model));
+            });
```

Comment:
Better keep it flat:

```suggestion
        }).then(function() {
            return updateModel(state);
        }).then(function(model) {
            history.pushState(null, null, '#!list');
            return onNavigate(null, Object.assign({}, state, model));
        });
```

Reply:
Also `extractJSON()` is relevant because it also checks that the response was successful (not obvious from the name, I know).

Reply:
> Better keep it flat
Sorry, I am not sure how to do this correctly. What would the almost same lines in `onSubmit` look like if we keep it flat there as well? It seems like there are too many brackets now in front of `.catch`

Reply:
I think I got it. Do you want me to change it in `onSubmit` as well?

Reply:
I don't think this would work in `onSubmit()`. If we use the flat structure, each function only has access to the result from its direct predecessor:

```javascript
return fetch('api.php', {
    method: 'POST',
    body: JSON.stringify(data),
    credentials: 'same-origin',
}).then(extractJSON).then(function(result) {  // result is defined here
    return updateModel(state)
}).then(function(model) {
    history.pushState(null, null, '#!detail/' + result.id);  // result is not available here
    return onNavigate(null, Object.assign({}, state, model));
}).catch(function(err) {
    console.error(err);
});
```

Reply:
Seems like it didn't get it after all ;) Thanks for the explanation!